exemplars_parent/exemplars/src/main/java/io/prometheus/client/exemplars/api/ExemplarConfig.java

```diff
@@ -0,0 +1,90 @@
+package io.prometheus.client.exemplars.api;
+
+import io.prometheus.client.exemplars.impl.DefaultExemplarSampler;
+import io.prometheus.client.exemplars.impl.NoopExemplarSampler;
+import io.prometheus.client.exemplars.tracer.common.SpanContext;
+import io.prometheus.client.exemplars.tracer.otel.OpenTelemetrySpanContext;
+import io.prometheus.client.exemplars.tracer.otel_agent.OpenTelemetryAgentSpanContext;
+
+/**
+ * Static configuration class for Exemplar behavior.
+ */
+public class ExemplarConfig {
+
+  private static final ExemplarConfig instance = new ExemplarConfig();
+
+  private final ExemplarSampler noopExemplarSampler;
+  private final ExemplarSampler defaultExemplarSampler;
+
+  private volatile HistogramExemplarSampler histogramExemplarSampler;
+  private volatile CounterExemplarSampler counterExemplarSampler;
+
+  private ExemplarConfig() {
+    noopExemplarSampler = new NoopExemplarSampler();
+    SpanContext spanContext = findSpanContextSupplier();
+    if (spanContext != null) {
+      defaultExemplarSampler = new DefaultExemplarSampler(spanContext);
+    } else {
+      defaultExemplarSampler = noopExemplarSampler;
+    }
+    counterExemplarSampler = defaultExemplarSampler;
+    histogramExemplarSampler = defaultExemplarSampler;
+  }
+
+  // Counter
+
+  public static void setCounterExemplarSampler(CounterExemplarSampler counterExemplarSampler) {
+    if (counterExemplarSampler == null) {
+      throw new NullPointerException();
+    }
+    instance.counterExemplarSampler = counterExemplarSampler;
+  }
+
+  // Histogram
+
+  public static void setHistogramExemplarSampler(HistogramExemplarSampler histogramExemplarSampler) {
+    if (histogramExemplarSampler == null) {
+      throw new NullPointerException();
+    }
+    instance.histogramExemplarSampler = histogramExemplarSampler;
+  }
+
+  // Disable all of the above
+
+  public static void disableExemplars() {
+    instance.counterExemplarSampler = instance.noopExemplarSampler;
+    instance.histogramExemplarSampler = instance.noopExemplarSampler;
+  }
+
+  // Methods for internal use
```

Comment:
If they're internal, why are they public?

Reply:
I removed the comment. It makes sense to have these as public methods, because a user might want to keep a copy of the original default `ExemplarSampler`.